src/script/entity/message/SystemMessage.ts

```diff
@@ -23,16 +23,20 @@ import {Message} from './Message';
 import {RenameMessage} from './RenameMessage';
 
 import {SuperType} from '../../message/SuperType';
-import {SystemMessageType} from '../../message/SystemMessageType';
+import {SystemMessageIcon, SystemMessageType} from '../../message/SystemMessageType';
 
 export class SystemMessage extends Message {
   public caption: ko.PureComputed<string>;
   public system_message_type: SystemMessageType;
+  public icon: SystemMessageIcon | null;
+  public includeSenderName: boolean;
```

Comment:
Although the idea of making the component as dumb as possible, I think this implementation has some layering problems. 

Here we are in a pure logic class that should not have a single mention of UI layer parts. 
`icon` and `includeSenderName` are intents that are only for the UI (to which this class should not care about). 

Here a `type` property would be the right semantic. 
For the particular display logic, depending on the `type` we could have sub-components of the `SystemMessage` that renders differently according to the `type`

WDYT?